src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int HOUR_INDEX = 1;
+    private static final int SALARY_INDEX = 2;
+    private static final int DATE_INDEX = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateStart = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate dateEnd = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+
+        StringBuilder result = new StringBuilder().append("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name: names) {
+            int resultSalary = 0;
+            for (String line: data) {
+                String[] arrayOfData = line.split(" ");
+                LocalDate currentDate
+                        = LocalDate.parse(arrayOfData[DATE_INDEX], DATE_TIME_FORMATTER);
+                if (currentDate.compareTo(dateStart) > 0
+                        && currentDate.compareTo(dateEnd) < 0
+                        || currentDate.equals(dateStart)
+                        || currentDate.equals(dateEnd)) {
+                    if (line.contains(name)) {
```

Comment:
lets split line before this if, check with equals
Because some name can be a part of another name - it would lead to a mistake